python/tirex_tracker/archive_utils.py

```diff
@@ -55,13 +55,25 @@ def _create_notebook_zip_archive(
     with redirect_stdout(None):
         ipython.run_line_magic("save", f"-f {script_file_path} 1-9999")
         ipython.run_line_magic("notebook", str(notebook_file_path))
-    return _create_zip_archive(
+    ret = _create_zip_archive(
         metadata_directory_path=metadata_directory_path,
         base_directory_path=metadata_directory_path,
         script_file_path=script_file_path,
         notebook_file_path=notebook_file_path,
     )
 
+    script_file_path.unlink()
+    notebook_file_path.unlink()
```

Comment:
Why are the simple script and notebook files removed here?
I think it is often easier to use these directly instead of first unzipping the archive and then using the unzipped notebook/script.

Reply:
Because of the single source principle. After working a bit with this, I learned that it is super confusing when this is duplicated.

Reply:
Then let's directly use `TemporaryFile` or `NamedTemporaryFile` instead, which automatically deletes the temporary files. The code should then not need to rely on any predetermined file path